src/Tribe/Utils/Element_Attributes.php

```diff
@@ -0,0 +1,228 @@
+<?php
+namespace Tribe\Utils;
+
+/**
+ * Class Element_Attributes to handle HTML attributes for elements.
+ *
+ * @since  TBD
+ *
+ * @package Tribe\Utils
+ */
+class Element_Attributes {
+	/**
+	 * Store the results of parsing the attributes.
+	 *
+	 * @since  TBD
+	 *
+	 * @var array
+	 */
+	protected $results = [];
+
+	/**
+	 * Stores the arguments passed.
+	 *
+	 * @since  TBD
+	 *
+	 * @var array
+	 */
+	protected $arguments = [];
+
+	/**
+	 * Setups an instance of Element Attributes.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		$this->arguments = func_get_args();
+	}
+
+	/**
+	 * When invoked this class will return the full HTML attributes.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` attribute1="value1" attribute2="value2" `
+	 */
+	public function __invoke() {
+		$this->arguments = func_get_args();
+		return $this->get_attributes();
+	}
+
+
+	/**
+	 * When cast to string an instance will return the full HTML attributes.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` attribute1="value1" attribute2="value2" `
+	 */
+	public function __toString() {
+		return $this->get_attributes();
+	}
+
+	/**
+	 * Gets the full HTML attributes for this instance of Element Attributes.
+	 * It will contain a space on each end of the attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` attribute1="value1" attribute2="value2" `
+	 */
+	public function get_attributes() {
+		$attributes = $this->get_attributes_as_string();
+
+		// Bail with empty string when no attributes are present
+		if ( ! $attributes ) {
+			return '';
+		}
+
+		return " $attributes ";
+	}
+
+	/**
+	 * Gets a space separated string of all attributes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string
+	 */
+	public function get_attributes_as_string() {
+		return implode( ' ', $this->get_attributes_array() );
+	}
+
+	/**
+	 * Get the array of attributes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_attributes_array() {
+		$this->results = [];
+		$attributes    = [];
+
+		$this->parse_array( $this->arguments );
+
+		foreach ( $this->results as $key => $val ) {
+			if ( ! $val && '0' !== $val ) {
+				continue;
+			}
+
+			if ( is_bool( $val ) ) {
+				$attributes[] = esc_attr( $key );
+			} else {
+				$attributes[] = esc_attr( $key ) . '="' . esc_attr( $val ) . '"';
+			}
```

Comment:
My main concern here is around this and escaping the key + value. in `Element_Classes` we had a single attribute `class` and the function to sanitize html classes. I'm not sure what the best approach for attributes is here..